.github/workflows/gh-pages.yml

```diff
@@ -1,9 +1,10 @@
 name: Deploy to GitHub Pages
 
 on:
-  release:
-    types:
-      - "published"
+  push:
+    branches: [ main ]
+  pull_request:
```

Comment:
we shouldn't deploy on PR. Only if `main` is updated

Reply:
this is causing the workflow to fail. Random PRs should **not** have permissions to deploy to our github pages. Anyone who can open a PR could then do anything. The workflow should work on main

Reply:
ahhh right. I thought otherwise PRs for docs wouldn't go through... But of course, this just gets run as soon as the PR is lodged. GREAT catch